cmd/history.go

```diff
@@ -85,27 +86,31 @@ func displayUpdate(updates []backend.UpdateInfo, opts display.Options) {
 		}
 		fmt.Printf("Message: %v\n", update.Message)
 
-		printResourceChanges(colors.GreenBackground, colors.Black, "+", update.ResourceChanges["create"], colors.Reset)
-		printResourceChanges(colors.RedBackground, colors.Black, "-", update.ResourceChanges["delete"], colors.Reset)
-		printResourceChanges(colors.YellowBackground, colors.Black, "~", update.ResourceChanges["update"], colors.Reset)
-		printResourceChanges(colors.BlueBackground, colors.Black, " ", update.ResourceChanges["same"], colors.Reset)
+		printResourceChanges(colors.GreenBackground, colors.Black, "+", colors.Reset, update.ResourceChanges["create"])
+		printResourceChanges(colors.RedBackground, colors.Black, "-", colors.Reset, update.ResourceChanges["delete"])
+		printResourceChanges(colors.YellowBackground, colors.Black, "~", colors.Reset, update.ResourceChanges["update"])
+		printResourceChanges(colors.BlueBackground, colors.Black, " ", colors.Reset, update.ResourceChanges["same"])
 
 		timeStart := time.Unix(update.StartTime, 0)
 		timeCreated := humanize.Time(timeStart)
 		timeEnd := time.Unix(update.EndTime, 0)
 		duration := timeEnd.Sub(timeStart)
-		fmt.Printf("%2sUpdated %s took %s\n", "", timeCreated, duration)
+		fmt.Printf("%sUpdated %s took %s\n", " ", timeCreated, duration)
 
-		empty := func(s string) bool {
+		isEmpty := func(s string) bool {
 			return len(strings.TrimSpace(s)) == 0
 		}
-
-		if len(update.Environment) != 0 {
-			indent := 4
-			if !empty(update.Environment["github.login"]) && !empty(update.Environment["git.committer"]) && !empty(update.Environment["git.committer.email"]) && !empty(update.Environment["git.head"]) {
-				fmt.Printf("%*sGithub-Login: %s\n", indent, "", update.Environment["github.login"])
-				fmt.Printf("%*sGithub-Committer: %s <%s>\n", indent, "", update.Environment["git.committer"], update.Environment["git.committer.email"])
-				fmt.Print(opts.Color.Colorize(fmt.Sprintf("%*s%scommit %s%s\n", indent, "", colors.Yellow, update.Environment["git.head"], colors.Reset)))
+		var keys []string
+		for k := range update.Environment {
+			keys = append(keys, k)
+		}
+		sort.Strings(keys)
+		indent := 4
+		for _, k := range keys {
+			if k == backend.GitHead && !isEmpty(update.Environment[k]) {
+				fmt.Print(opts.Color.Colorize(fmt.Sprintf("%*s%s%s: %s%s\n", indent, "", colors.Yellow, k, update.Environment[k], colors.Reset)))
```

Comment:
Nit: If you ever need to deeply nest function calls (`fmt.Print(opts.Color.Colorize(fmt.Sprintf(`) IMHO it is preferable to just introduce local variables, as that would make it much easier to read.

```
m := fmt.Sprintf("%*s%s%s: %s%s\n", indent, "", colors.Yellow, k, update.Environment[k], colors.Reset)
colorized := ops.Color.Colorize(m)
fmt.Print(colorized)

or just  move the most complicated one out.
m := fmt.Sprintf("%*s%s%s: %s%s\n", indent, "", colors.Yellow, k, update.Environment[k], colors.Reset)
fmt.Print(ops.Color.Colorize(m)))
```